### PR TITLE
fix(TDP-4142): PE - fix ui glitches 

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4142

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/505/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/505/)

File: dataprep-webapp/src/app/components/export/_export.scss [hunk 1]
```diff
@@ -14,16 +14,19 @@
...
-		cursor: not-allowed;
-	}
 	.btn {
 		height: $export-btn-height;
 		min-height: $export-btn-height;
 	}
 	li > a {
 		width: 100%;
+
+		&.disabled:hover {
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/_export.scss]: When disabled, a dash appears at the end of the export type : `<span ng-if="!exportType.enabled"> - </span>`. Is it normal ?

image

File: dataprep-webapp/src/app/components/export/_export.scss [hunk 2]
```diff
@@ -14,16 +14,19 @@
...
+			cursor: not-allowed;
+		}
+
+		&:not(.disabled) {
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/_export.scss]: I think you should define this color at top level and override it by the lighter one only when disabled.

File: dataprep-webapp/src/app/components/folder/selection/folder-tree-node/_folder-tree-node.scss [hunk 3]
```diff
@@ -35,6 +35,10 @@ $folder-tree-node-btn-click-area: 10px !default;
 			border-left: 6px solid transparent
 		}
 
+		&:not(.caret) {
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/folder/selection/folder-tree-node/_folder-tree-node.scss]: :ok_hand:

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 1 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/export/_export.scss]: Yes but the backend should send a "disableReason" to explain why it is disabled

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 2 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/export/_export.scss]: the problem is that all the li>a of the subheader uses lighter color

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 1 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/_export.scss]:
```
li > a {
    width: 100%;
    color: $dark-gray;

    &.disabled {
        color: $lighter-one;
        ...
    }
...
```

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 2 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/_export.scss]: I mean : 

```
li > a {
    width: 100%;
    color: $dark-gray;

    &.disabled {
        color: $lighter-one;
        ...
    }
...
```

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 1 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/_export.scss]: Oh, ok, my bad.

File: dataprep-webapp/src/app/components/export/_export.scss (hunk 2 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/export/_export.scss]: ok

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/514/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/514/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/516/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/516/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/522/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/522/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/523/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/523/)

**Comment by ncomont** [comment]
cherry picked on maitenance/2.1

**Pull request merged**